libsolidity/interface/CompilerStack.cpp

```diff
@@ -492,7 +499,12 @@ eth::AssemblyItems const* CompilerStack::assemblyItems(string const& _contractNa
 eth::AssemblyItems const* CompilerStack::runtimeAssemblyItems(string const& _contractName) const
 {
 	if (m_stackState != CompilationSuccessful)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Compilation was not successful."));
+	{
+		if (m_parserErrorRecovery)
```

Comment:
I'm still of the opinion that this has to produce an exception.

Reply:
Throws an exception now.